ROOT Head fails to compile on macOS(11, 12) since Jan 27, 2023

### Describe the bug
In the lcgcmake build system
Compiling ROOT master on macOS 11 (http://cdash.cern.ch/viewFiles.php?buildid=218660) or 12 (http://cdash.cern.ch/viewFiles.php?buildid=218674) fails to compile with the error message

```
[ 55%] Copying clang's __clang_hip_libdevice_declares.h...
In file included from /Users/sftnight/build/workspace/lcg_nightly_pipeline/build/projects/ROOT-HEAD/src/ROOT/HEAD/core/macosx/src/TMacOSXSystem.mm:12:
/Users/sftnight/build/workspace/lcg_nightly_pipeline/build/projects/ROOT-HEAD/src/ROOT/HEAD/core/macosx/inc/TMacOSXSystem.h:16:10: fatal error: 'TUnixSystem.h' file not found
[ 55%] Linking CXX static library ../../libLLVMDebugInfoCodeView.a
#include "TUnixSystem.h"
```

Seems related to #10690 Which was touching the related CMakeLists.txt files

### Expected behavior

Root compiles successfully on macOS

### To Reproduce
<!--
Steps to reproduce the behavior:
1. Your code that triggers the issue: at least a part; ideally something we can run ourselves.
2. Don't forget to attach the required input files!
3. How to run your code and / or build it, e.g. `root myMacro.C`, ...
-->

### Setup

1. ROOT version: master
2. Operating system: macOS 11, 12
3. git clone

--- a/core/macosx/CMakeLists.txt
+++ b/core/macosx/CMakeLists.txt
@@ -29,6 +29,7 @@ if(cocoa)
   )
 
   target_compile_options(Macosx PRIVATE -ObjC++)
+  target_include_directories(Macosx PRIVATE ../unix/inc)
 
   target_sources(Core PRIVATE $<TARGET_OBJECTS:Macosx>)
 

